subsys/spm/spm.c

```diff
@@ -271,10 +271,13 @@ static int spm_config_peripheral(u8_t id, bool dma_present)
 			PERIPH_LOCK;
 	}
 
-	/* Even for non-present peripherals we force IRQs to be routed
-	 * to Non-Secure state.
-	 */
-	irq_target_state_set(id, 0);
+	/* Route IRQs for peripherals which has it */
+	if (id != NRFX_PERIPHERAL_ID_GET(NRF_P0)) {
```

Comment:
Can we do something generalized here?

nRF5340 and nRF9160 have both P0 and P1, so this is not sufficient, in the first place.

Also, it's important that the IRQ line exists, otherwise the underlying Zephyr function will ASSERT.

So before calling irq_target_state_set, we need to make sure the IRQ NVIC line exists. I would prefer a solid way of checking this; i.e. not relying in "knowing" that NRF_P0 does not have IRQ wired.

We can do this in SPM or provide an ARM-only API for the upstream project - what do you think?

Reply:
I'm not aware of any method of going from peripheral ID to checking if the IRQ line exits, but I guess we could re-use the code which actually triggered the assert. 

I want this to be as far upstream as possible, and as generic as possible :) IMO it makes sense to have a function which allows to you ensure that `irq_target_state_set` won't assert. 

perhaps checking `((uint32_t)(((NVIC->ITNS[(((uint32_t)IRQn) >> 5UL)] & (1UL << (((uint32_t)IRQn) & 0x1FUL))) != 0UL)` like `NVIC_SetTargetState`